infra/discovery/src/lib/search.js

```diff
@@ -84,6 +84,55 @@ class Listing {
     return listingId
   }
 
+  /**
+   * Gets a listing
+   * @param {string} id
+   */
+  static async get(id) {
+    return client.get({
+      index: LISTINGS_INDEX,
+      type: LISTINGS_TYPE,
+      id: id
+    })
+  }
+
+  /**
+   *
+   * @param {string} id
+   * @param {string[]} scoreTags
+   */
+  static async updateScoreTags(id, scoreTags) {
+    const result = await client.get({
+      index: LISTINGS_INDEX,
+      type: LISTINGS_TYPE,
+      id: id
+    })
+    const listing = result._source
+    listing.scoreTags = scoreTags
+    const { scoreMultiplier } = await scoring.scoreListing(result._source)
+    listing.scoreMultiplier = scoreMultiplier
+    const body = {
```

Comment:
nit: add some comments here or at top of function to explain we use painless script to re-index the document with a new score calculated based on scoreTags.
Is painless actually necessary rather than just re-indexing the doc we got from result._source ?

Reply:
Comment added. Painless isn't necessary, but it seems cleaner and reduces the size of the race condition window when a doc could be being updated simultaneously by multiple systems.